app/client/src/store/getters.ts

```diff
@@ -0,0 +1,15 @@
+import { AnalysisType } from '@/types';
+import { RootState } from '@/store/state';
+
+export default {
+  analysisProgress(state: RootState) {
```

Comment:
We'd normally use the `GetterTree` vuex interface for getters.  [Example here](https://github.com/mrc-ide/wodin/blob/main/app/static/src/app/store/modelFit/getters.ts). Not sure if that would help with your type issues when mapping from the component though.